tests/test_utils/assert_metrics_dict.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Optional
+
+from pytest import approx
+
+DEFAULT_TOLERANCE = 0.0005
+
+
+def assert_metrics_dict(metrics_to_assert: dict[str, Any], metrics_saved: dict[str, Any]) -> list[str]:
+    errors = []
+
+    def _assert(value: Any, saved_value: Any) -> Optional[str]:
```

Comment:
Any reason to make this an inline function rather than popping it out of here into it's own separate function above this method?

Reply:
I didn't write this test. A lot of of the code in this pr was previously in the main branch, I'm not sure why it's showing it as having been added by me

Reply:
Lol not sure if you'll get my last comment but I actually did write this. I forgot because I just copied it from the smoke tests. My bad. But yeah it was an inline function because I just copy pasted the method from the smoke tests

Reply:
I will clean it up